core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -1958,6 +1958,7 @@ SDDkwd__(EXE_DIAGNOSTIC_EVENTS,		"OFF"),
 
   DDkwd__(HIVE_DEFAULT_CHARSET,            (char *)SQLCHARSETSTRING_UTF8),
   DD_____(HIVE_DEFAULT_SCHEMA,                  "HIVE"),
+  DD_____(HIVE_FILE_CHARSET,            (char *)SQLCHARSETSTRING_UTF8),
```

Comment:
If we have to join two Hive tables, one in UTF8 and the other in GBK, I suppose this method will have to be extended? Does Hive metadata store charset info? If not would a comment next to the tablename help? This is just a minor suggestion for something you may have already considered, not a request to add this now.